domain/machine/state/state_test.go

```diff
@@ -769,3 +769,30 @@ func (s *stateSuite) TestSetKeepInstanceNotFound(c *gc.C) {
 	err := s.state.SetKeepInstance(context.Background(), "666", true)
 	c.Assert(err, jc.ErrorIs, machineerrors.MachineNotFound)
 }
+
+func (s *stateSuite) TestSetLXDProfiles(c *gc.C) {
+	err := s.state.CreateMachine(context.Background(), "666", "", "deadbeef")
+	c.Assert(err, jc.ErrorIsNil)
+	err = s.state.SetLXDProfiles(context.Background(), "deadbeef", []string{"profile1", "profile2"})
+	c.Assert(err, jc.ErrorIsNil)
+
+	profiles, err := s.state.LXDProfiles(context.Background(), "deadbeef")
```

Comment:
**todo:** IMHO we should have separate tests for both insertion and set, with sql query to check state after insertion, and sql query to setup state before select.

It make the tests independant. However, we can keep `state.CreateMachine` in arrange step, because as far I remember, setting a machine is not trivial (but we could have an helper function laying in test to setup a simple machine with raw sql)

We are actually testing that we are inserting stuff in DB, which is the implementation of state.

Reply:
Yeah right, I usually do that but this time it was so simple that thought no need. Done it in a2920161dc7071233e891ffb74d360088da68463.